src/main/java/com/bootcamp/demo/model/Booking.java

```diff
@@ -0,0 +1,98 @@
+package com.bootcamp.demo.model;
+
+import java.util.Objects;
+import java.util.UUID;
+
+public final class Booking {
+    private String id;
+    private String serialNumber;
+    private long accountId;
+    private String startDate;
```

Comment:
Check this out. I think this resolves the problem with dates. [link](https://www.baeldung.com/spring-date-parameters)

Reply:
You could as well use long datatype and store the value as millis and convert it in the application via something like:
`Instant.ofEpochSecond(value).atZone(ZoneId.systemDefault()).toLocalDateTime();`
There is also the possibility to create a custom document mapper which would map from QueryDocumentSnapshot to an internal model class.